core-bundle/src/Image/Studio/FigureBuilder.php

```diff
@@ -265,6 +269,33 @@ public function from(FilesModel|ImageInterface|int|string|null $identifier): sel
         return $this->fromPath($identifier);
     }
 
+    /**
+     * Sets the image resource from a path inside a VFS storage.
+     */
+    public function fromStorage(VirtualFilesystemInterface $storage, string|Uuid $location): self
+    {
+        try {
+            $stream = $storage->readStream($location);
+        } catch (VirtualFilesystemException|UnableToResolveUuidException $e) {
+            $this->lastException = new InvalidResourceException(sprintf('Could not read resource from storage: %s', $e->getMessage()), previous: $e);
+
+            return $this;
+        }
+
+        // TODO: Our image libraries cannot deal with streams, yet. However,
+        //       in case of STDIO streams, we can already be forward compatible
+        //       by reading the local file path from stream's meta data.
+        $metadata = stream_get_meta_data($stream);
+
+        if ('STDIO' !== ($type = $metadata['stream_type'])) {
```

Comment:
Should we check for `wrapper_type` being `plainfile` here too?

Reply:
And probably also check `Path::isAbsolute($metadata['uri'])`?

Reply:
Sounds like a sensible idea :+1: 
Done in e72484d910b2afcc3bdb362e7a3b412c3dc6e0a1.

Reply:
> And probably also check Path::isAbsolute($metadata['uri'])?

Are you sure relative paths would even be possible?

Reply:
The docs state `URI/filename` and I saw some comments about this not always being a path. Not sure if this can happen here, but I’d prefer to be extra cautious ☺️

Reply:
Yeah, it can indeed be relative, e.g. when you have something like `$stream = fopen('my-file', 'rb');`

Adjusted in  185f449c3daa62a7ff530cb1ccd29c415267d759.